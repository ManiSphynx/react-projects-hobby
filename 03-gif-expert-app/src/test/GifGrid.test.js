import React from "react";
import { shallow } from "enzyme";
import { GifGrid } from "../components/GifGrid";
import { useFetchGifs } from "../hooks/useFetchGifs";
jest.mock("../hooks/useFetchGifs");

describe("Pruebas en GifGrid", () => {
  const category = "Categoria";
  test("Probar que GifGrid se muestre correctamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("debe de mostrar items cuando se cargan las imagenes useFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "http://localhost/cualquier/cosa.jpg",
        title: "Hola mundo",
      },
      {
        id: "123",
        url: "http://localhost/cualquier/co785475sa.jpg",
        title: "Hola mundo",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
