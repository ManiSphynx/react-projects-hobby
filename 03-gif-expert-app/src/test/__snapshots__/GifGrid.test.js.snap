// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Probar que GifGrid se muestre correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__delay-1s"
  >
    Categoria
  </h3>
  <p
    className="animate__animated animate__flash animate__delay-1s"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en GifGrid debe de mostrar items cuando se cargan las imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__delay-1s"
  >
    Categoria
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Hola mundo"
      url="http://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Hola mundo"
      url="http://localhost/cualquier/co785475sa.jpg"
    />
  </div>
</Fragment>
`;
